### PR TITLE
Rewrite jp2 create robot to handle files not being present.

### LINKED ISSUE #1090: Targeted file updates in Jp2 create
In targeted file updates, the tiff file will not exist in the workspace if it has not changed. The Jp2Create robot expects it to exist so that a JP2 file can be generated at the cocina model updated.

Possible approaches:
* Ignore, tiff files will not be accessioned with Pre-assembly
* Require tiff files to be present so that Jp2Create can be run (i.e., they cannot be used with targeted file updates)
* In Pre-assembly, if a tiff file has not changed then add the cocina file for the JP2 into the structural. When the tiff file is not present, but there is a cocina file for the Jp2 in the structural, the Jp2Create robot does not attempt to generate a new Jp2.

Also needs to be considered how each of these approaches will affect shelving, which probably assumes that the Jp2 file is present.

Thoughts @andrewjbtw ?

### PR DESCRIPTION
**Pull request by justinlittman**
closes #1090

## Why was this change made? 🤔
Deciding how, when to create JP2s is very complex

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, including writes to shared file systems, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

Unit, integration

### CODE REVIEW

**Comment by edsu** [comment]
I think this needs a rebase w/ main to bring in latest Cocina updates?

**Review by lwrubel — APPROVED** [review]
Approved, but with so much complicated logic, you might want another pair of eyes on it before merging.

File: lib/robots/dor_repo/assembly/jp2_create.rb
Comment (ndushay) [inline lib/robots/dor_repo/assembly/jp2_create.rb:57]: would it make sense to check for the cocina_jp2 file before some of these steps?

File: lib/robots/dor_repo/assembly/jp2_create.rb
Comment (ndushay) [inline lib/robots/dor_repo/assembly/jp2_create.rb:51]: how is the loop working if the loop is on cocina_files, but we check against all cocina_files to find a single cocina_jp2_file?

File: lib/robots/dor_repo/assembly/jp2_create.rb
Comment (ndushay) [inline lib/robots/dor_repo/assembly/jp2_create.rb:59]: files does not exist <-- typo

**Review by ndushay — CHANGES_REQUESTED** [review]
i have some points of confusion and a suggestion.

File: lib/robots/dor_repo/assembly/jp2_create.rb
Comment (justinlittman) [reply lib/robots/dor_repo/assembly/jp2_create.rb:51]: This is all of the cocina files in a file set, not all cocina files.

File: lib/robots/dor_repo/assembly/jp2_create.rb
Comment (justinlittman) [reply lib/robots/dor_repo/assembly/jp2_create.rb:57]: It's another option, that I'm not sure that any formulation of this logic is going to be clearer than any other. If you had a formulation, I'd be happy to consider.

File: lib/robots/dor_repo/assembly/jp2_create.rb
Comment (justinlittman) [reply lib/robots/dor_repo/assembly/jp2_create.rb:59]: Yes, yes it is. Fixed.

File: lib/robots/dor_repo/assembly/jp2_create.rb
Comment (ndushay) [reply lib/robots/dor_repo/assembly/jp2_create.rb:51]: I mean the cocina_files object on line 31 and line 34, which is stepping through them, is also looking at the cocina_files same object on line 51 ... inside the loop.

File: lib/robots/dor_repo/assembly/jp2_create.rb
Comment (ndushay) [reply lib/robots/dor_repo/assembly/jp2_create.rb:57]: backing away slowly.  There might be a way but I'm not going to spend the time on it.  If someone wants, they can refactor it later.  Working code wins.

**Review by ndushay — APPROVED** [review]

**Pull request merged**
